content/dns-lookup-api.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "DNS lookup API"
+date: 2018-03-29T14:49:22+02:00
+---
+
+An API that allows a WebExtension to lookup for arbitrary DNS records in highly
+efficient manner.
+
+## Motivation
+
+Use DNS lookup to create a bridge between centralized and decentralized technologies.
+
+There are two DNS resource record types that are relevant to this objective:
+
+- `SRV` records provide means for specifying the location of services ([RFC 2782][2]).
+- `TXT` records provide the ability to associate arbitrary string metadata with a host ([RFC 1464](https://tools.ietf.org/html/rfc1464)).
+
+`TXT` records are much more versatile. There are historical examples of
+[publishing PGP pubkeys][3], [domain ownership verification][4],
+['serverless' redirect services][5] and more.
+
+IPFS uses `TXT` record for publishing [dnslink][6] as a means of exposing content
+from IPFS Path under `/ipns/${fqdn}/` namespace. Validation of `/ipns/` paths
+includes DNS lookup to verify if `/ipns/${fqdn}` is backed by the presence of
+dnslink `TXT` record. 
+
+Without dedicated API for DNS lookups browser extensions are forced to use
+third-party [DNS-over-HTTPS][7] services.
+This workaround comes at a price:
+
+- Dependency on hardcoded third party lookup service introduces the single
+  point of failure.  It also makes MITM attacks easier and increases
+  probability of leaking private information.
+
+- Sending HTTP GET for each query is much slower than native DNS client already
+ present in web browser. Overhead makes it not feasible to run
+ `TXT` lookup too often or during time-critical paths such as
```

Comment:
Overhead makes it not feasible to run `TXT` lookup too often or during time-critical paths...
=>
The overhead is particularly undesirable during time-critical paths...